src/GraphODataTemplateWriter/Settings/ConfigurationService.cs

```diff
@@ -11,18 +11,27 @@ public static class ConfigurationService
     {
         private static IConfigurationProvider _configurationProvider;
         private static TemplateWriterSettings templateWriterSettings = null;
-
-        public static void Initialize(IConfigurationProvider configurationProvider)
+        private static string targetLanguage = null;
+        public static void Initialize(IConfigurationProvider configurationProvider, string targetLanguage = null)
         {
             _configurationProvider = configurationProvider;
+            if (!String.IsNullOrEmpty(targetLanguage))
+            {
+                ConfigurationService.targetLanguage = targetLanguage;
+            }
         }
 
         private static TemplateWriterSettings LoadSettingsForLanguage()
         {
             TemplateWriterSettings mainTWS = _configurationProvider.GetConfiguration<TemplateWriterSettings>();
```

Comment:
Is the default language really CSharp or is what was last checked in to the template writer settings? I propose that we make CSharp the default, and ignore template writer settings.

Reply:
The options class has the default of CSharp so targetLanguage should never be empty.  So we should never use the value in the settings file.